### PR TITLE
Check invalid protocol

### LINKED ISSUE #33: Invalid protocol produce app error
Kdyz zadam v `Base Url` adresu s nesmyslnym protokolem na zacatku tak extractor skonci app errorem:

Priklad: `fake://keboola.com`

`docker-0264d4d1b9717ddac912423f0f55ee48`

Podle logu to vyhodilo `GuzzleHttp\Exception\RequestException` To bych cekal ze se prevede na user error vzdycky.

### PR DESCRIPTION
**Pull request by ondrajodas**
Fixes #33

### CODE REVIEW

File: src/HttpExtractor.php [hunk 1]
```diff
@@ -33,6 +33,14 @@ public function __construct(
 
     public function extract(UriInterface $httpSource, string $filesystemDestination): void
     {
+        // check is valid URL
+        if (!preg_match('/^https?/', (string) $httpSource)) {
+            throw new UserException(sprintf(
+                'This url "%s" is not valid',
```
Comment (tomasfejfar) [inline src/HttpExtractor.php]: `Url "%s" is not valid `. Ale ještě lépe by bylo vytahnout match z toho regexu a vyhodit rovnou `Protocol "%s" is not valid. Only http and https are allowed`. 

Plus ten regex by měl být kompletní IMHO. `/^https?:\/\//`. Takhle je `httpsx://example.com` platný. 

Každopádně ale ten problém byl v tom, že když je "nějak" neplatné URL, tak to vyhodí application error. Takže spíš než validovat to URL by bylo potřeba chytit tu výjimku a vyhodit UserException s odpovídající chybou. 

Kód tam na to už je a předpokládám, že neplatný protokol bude `CURLE_UNSUPPORTED_PROTOCOL`.

Comment (tomasfejfar) [reply src/HttpExtractor.php]: Případně by ten regex mohl být rovnou v config definition - https://symfony.com/doc/current/components/config/definition.html#validation-rules (ifTrue(function ($v) {regex}) thenInvalid()

**Review by tomasfejfar — CHANGES_REQUESTED** [review]
Ještě k samotnému PR - do description se snažíme psát odkaz na to, co to fixuje. Typicky `Fixes #123456` - což udělá pak to, že to automaticky zavře tu issue, jakmile je PR mergnutý.

File: src/HttpExtractor.php (hunk 1 above)
Comment (ondrajodas) [reply src/HttpExtractor.php]: přehodil jsem ten regex do config definition + udělal na to test

**Comment by ondrajodas** [comment]
> Ještě k samotnému PR - do description se snažíme psát odkaz na to, co to fixuje. Typicky `Fixes #123456` - což udělá pak to, že to automaticky zavře tu issue, jakmile je PR mergnutý.

ok, doplnil jsem to tam

**Review by tomasfejfar — CHANGES_REQUESTED** [review]
Poslal jsem test, který failuje a na masteru projde.

**Comment by ondrajodas** [comment]
> Poslal jsem test, který failuje a na masteru projde.

poslal jsem tam opravu, teď by mělo být možné zadat i URL adresu bez protokolu

**Review by tomasfejfar — CHANGES_REQUESTED** [review]
Sorry, ještě další edge case. Nejsem si jistý, jestli by fakt nebylo lepší to nechat projít a jen zachytit tu exception s CURLE_*

**Comment by ondrajodas** [comment]
poslal jsem ještě jednu opravu, kde by se mělo brát cokoli co je před :// (pokud tam je) a to zkontrolovat 
a projít by mělo jen http(s)
pokud by tě ještě něco napadlo tak bych to potom popřípadě udělal přes to `CURLE_UNSUPPORTED_PROTOCOL`

**Review by tomasfejfar — APPROVED** [review]
Teď už to vypadá dobře :) 

Takže si můžeš i úplně dodělat to kolečko. Tj. mergnout, tagnout nový release v https://github.com/keboola/http-extractor/releases/new a až proběhne build, tak za 10-15minut se to syncne do KBC. Pak můžeš udělat i ověření přímo tam, že když si ho uděláš konfiguraci http extractoru, takže že fakt správně failne pro ty casy co jsi fixoval...  kdybys potřeboval pomoct, tak pingni...
